core/src/main/java/org/infinispan/remoting/transport/jgroups/SecurityActions.java

```diff
@@ -0,0 +1,36 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import org.infinispan.security.Security;
+import org.infinispan.security.actions.GetSystemPropertyAsBooleanAction;
+import org.infinispan.security.actions.GetSystemPropertyAsIntegerAction;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * Privileged actions for the package
+ *
+ * Do not move. Do not change class and method visibility to avoid being called from other
+ * {@link java.security.CodeSource}s, thus granting privilege escalation to external code.
+ *
+ * @author Dan Berindei
+ * @since 8.2
+ */
+final class SecurityActions {
+   private static <T> T doPrivileged(PrivilegedAction<T> action) {
+      if (System.getSecurityManager() != null) {
+         return AccessController.doPrivileged(action);
+      } else {
+         return Security.doPrivileged(action);
```

Comment:
On second thought I am not sure if we need this for simple system property retrieval...  can you confirm for sure @tristantarrant ?  It seems that system property will never check the authorized variable in Security when it is performed.